### PR TITLE
Feat/224 recipe videos

### LINKED ISSUE #224: feat(recipes): implement recipe videos feature
The video is not shown when I insert a YouTube link

### PR DESCRIPTION
**Pull request by edbertdai**
Implemented recipe video field in form. Accepts any valid YouTube video link.
The field data is converted into a YouTube embed link of the form `https://youtube.com/embed/{id}`
Then, the data gets written to `recipes/{index}/metadata/video` in the database.
The user is alerted if the entry is invalid and can only submit if the field is empty or a valid link.

Also implemented recipe video on recipe details page.
There is an `iframe` that defaults to `display: none` unless `metadata/video` is non-empty.
If `metadata/video` is non-empty then the `iframe` will be `display: block` and `src` set to the embed link.

Closes #224

### CODE REVIEW

File: source/components/recipe-contribute/recipe-contribute.js [hunk 1]
```diff
@@ -110,6 +111,12 @@ class RecipeContribute extends YummyRecipesComponent {
         const isFormValid = formElement.checkValidity();
         formElement.reportValidity();
 
+        // If video is invalid then alert user and do not continue
+        if (!this.checkVideoValidity()) {
+          alert("Video link is not valid");
+          return;
+        }
+
```
Comment (jbutera94) [inline source/components/recipe-contribute/recipe-contribute.js:119]: I think we can remove this in favor of the native HTML validation described in the comment.

File: source/components/recipe-contribute/recipe-contribute.js [hunk 2]
```diff
@@ -214,6 +225,12 @@ class RecipeContribute extends YummyRecipesComponent {
...
```
Comment (jbutera94) [inline source/components/recipe-contribute/recipe-contribute.js:233]: Same as above, we can remove this in favor of the native HTML validation in the comment.

**Review by jbutera94 — CHANGES_REQUESTED** [review]
Looks good, but I think we could utilize the native HTML validation described here: https://developer.mozilla.org/en-US/docs/Learn/Forms/Form_validation

This is how we currently check and display validation errors, and it's native to the browser, so I think adding the following event handler under line 60 (in `setupElement`) would allow a custom native error message for the validity of the youtube URL. Setting `setCustomValidity` to any error message will make the native validation error box appear, while setting `setCustomValidity` to a blank string is the same as saying it's a valid string.

```javascript
// Add custom validation to video URL field
const videoUrlInput = this.shadowRoot.getElementById("input-video-url");
videoUrlInput.addEventListener("input", () => {
  if (!this.checkVideoValidity()) {
    videoUrlInput.setCustomValidity("Video link is not valid.");
  } else {
    videoUrlInput.setCustomValidity("");
  }
});
```

File: source/components/recipe-contribute/recipe-contribute.js (hunk 1 above)
Comment (edbertdai) [reply source/components/recipe-contribute/recipe-contribute.js]: Removed

File: source/components/recipe-contribute/recipe-contribute.js (hunk 2 above)
Comment (edbertdai) [reply source/components/recipe-contribute/recipe-contribute.js]: Removed

**Review by jbutera94 — APPROVED** [review]
Looks good!

**Pull request merged**